flang/lib/Lower/Bridge.cpp

```diff
@@ -3772,9 +3797,13 @@ class FirConverter : public Fortran::lower::AbstractConverter {
       Fortran::lower::StatementContext localStmtCtx;
       hlfir::Entity rhs = evaluateRhs(localStmtCtx);
       hlfir::Entity lhs = evaluateLhs(localStmtCtx);
-      builder.create<hlfir::AssignOp>(loc, rhs, lhs,
-                                      isWholeAllocatableAssignment,
-                                      keepLhsLengthInAllocatableAssignment);
+      if (lhsIsDevice || rhsIsDevice) {
+        genCUDADataTransfer(builder, loc, lhsIsDevice, lhs, rhsIsDevice, rhs);
```

Comment:
Is it possible to have "non trivial" intrinsic assignments used in cuda transfer contexts: character assignments (require truncating/pading), derived type with allocatable components or derived type component with user defined assignments.

Is the fir.cuda_transfer representing/implementing these assignment aspects?